src/Microsoft.DotNet.Build.Tasks/GenerateTestOutputDirList.cs

```diff
@@ -0,0 +1,75 @@
+﻿using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Text;
+using System.Xml.Linq;
+
+namespace Microsoft.DotNet.Build.Tasks
+{
+    public class GenerateTestOutputDirList : Task
+    {
+        [Required]
+        public string OutputDirectory { get; set; }
+
+        [Required]
+        public ITaskItem[] TestOutputItems { get; set; }
+
+        [Required]
+        public string AssemblyName { get; set; }
+
+        public override bool Execute()
+        {
+
+            if (TestOutputItems.Length == 0)
+            {
+                throw new InvalidOperationException("TestOutputItems property is empty.");
+            }
+
+            if (!Directory.Exists(OutputDirectory))
+            {
+                Directory.CreateDirectory(OutputDirectory);
+            }
+
+            string testOutputDirList = Path.Combine(OutputDirectory, AssemblyName + ".txt");
+
+            string selectedDllPath = string.Empty;
+            foreach (var item in TestOutputItems)
+            {
+                selectedDllPath = item.ItemSpec.Replace('\\', '/');
+            }
+
+            HashSet<string> uniqueTestOutputDirectories = new HashSet<string>();
+            if (File.Exists(testOutputDirList))
+            {
+                using (StreamReader sr = new StreamReader(File.OpenRead(testOutputDirList)))
+                {
+                    string line;
+                    while ((line = sr.ReadLine()) != null)
+                    {
+                        uniqueTestOutputDirectories.Add(line);
+                    }
+                }
+            }
+
+            uniqueTestOutputDirectories.Add(selectedDllPath);
+            using (
+                FileStream fs = new FileStream(
+                    testOutputDirList,
+                    FileMode.Create,
+                    FileAccess.Write,
+                    FileShare.None))
+            using (StreamWriter listWriter = new StreamWriter(fs))
+            {
+                foreach (var item in uniqueTestOutputDirectories)
```

Comment:
You should sort these, so they don't come out in an essentially random order. HashSet doesn't return things in a predictable order.